torchaudio/datasets/cmudict.py

```diff
@@ -0,0 +1,92 @@
+        url (str, optional):
+            The URL to download the dictionary from.
+            (default: ``"http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b"``)
+        url_symbols (str, optional):
+            The URL to download the list of symbols from.
+            (default: ``"http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols"``)
+        download (bool, optional):
+            Whether to download the dataset if it is not found at root path. (default: ``False``).
+    """
+
+    def __init__(self,
+                 root: Union[str, Path],
+                 url: str = "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b",
+                 url_symbols: str = "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols",
+                 download: bool = False) -> None:
+
+        # Get string representation of 'root' in case Path object is passed
+        root = os.fspath(root)
+        if not os.path.isdir(root):
+            os.mkdir(root)
+
+        if download:
+            if not os.path.isdir(root):
+                checksum = _CHECKSUMS.get(url, None)
+                download_url(url, root, hash_value=checksum, hash_type="md5")
+                checksum = _CHECKSUMS.get(url_symbols, None)
+                download_url(url_symbols, root, hash_value=checksum, hash_type="md5")
+
+        self._root_path = root
+        basename = os.path.basename(url)
+        basename_symbols = os.path.basename(url_symbols)
+
+        with open(os.path.join(self._root_path, basename_symbols), "r") as text:
+            self._symbols = text.readlines()
+
+        with open(os.path.join(self._root_path, basename), "r") as text:
+            self._dictionary = self._parse_dictionary(text.readlines())
+
+        self._available_words = list(self._dictionary.keys())
+
+    def _parse_dictionary(self, lines: List[str]):
+        _alt_re = re.compile(r'\([0-9]+\)')
+        cmudict: List[Tuple[str, List[str]]] = list
+        for line in lines:
+            if len(line) and line[:3] != ";;;":  # ignore comments
+                parts = line.strip().split('  ')
+                # if a word have multiple pronunciations, there will be (number) appended to it
+                # for example, DATAPOINTS(1)
+                word = re.sub(_alt_re, '', parts[0])
+                phones = parts[1].split(" ")
+                cmudict.append(word, phones)
+        return cmudict
+
+    def __getitem__(self, n: int) -> Tuple[str, List[str]]:
+        """Load the n-th sample from the dataset.
+
+        Args:
+            n (int): The index of the sample to be loaded.
+
+        Returns:
+            tuple: The corresponding word and phonemes ``(word, [phonemes])``.
+
+        """
+        return self._dictionary[n]
+
+    def __len__(self) -> int:
+        return len(self._dictionary)
+
+    def get_symbols(self) -> List[str]:
+        return self._symbols
```

Comment:
What would happen if I do this?

```python
cmu_dict = CMUDict(root)
cmu_dict.get_symbols().append("foo")
print(cmu_dict.get_symbols())
```

Reply:
In Tedlium dataset, it uses `property` and `copy` to avoid in-place modification.

https://github.com/pytorch/audio/blob/d1d6dbc64a1dd7506ce6b2ff2f9969ae13a6b52d/torchaudio/datasets/tedlium.py#L180-L192

Reply:
Good point! Now it will call the copy method before returning ([here](https://github.com/pytorch/audio/pull/1627/files#diff-50ee900e0897ac7528dccd9bdd36d5f0c7cb2d45310aefb46a3bf281f73d88c2R102)).